google/resource_cloudfunctions_function_test.go

```diff
@@ -347,6 +353,21 @@ func testAccCloudFunctionsFunctionHasLabel(key, value string,
 	}
 }
 
+func testAccCloudFunctionsFunctionHasEnvironmentVariable(key, value string,
+	function *cloudfunctions.CloudFunction) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		val, ok := function.EnvironmentVariables[key]
+		if !ok {
+			return fmt.Errorf("Environment Variable with key %s not found", key)
+		}
+
+		if val != value {
+			return fmt.Errorf("Environment Variable value did not match for key %s: expected %s but found %s", key, value, val)
+		}
+		return nil
```

Comment:
You can refactor this to use the assignment in the if statement, e.g.

`if val, ok := function.EnvironmentVariables[key]; ok {
  if val != value {
    return fmt.Errorf(...)
  }
} else {
  return fmt.Errorf(...)
}